### PR TITLE
[mimictts] Use http method POST instead of GET

### PR DESCRIPTION
**Pull request by dalgwen**
Using POST method instead of GET allows longer request size.
(with get method, the request fails when the text is too long)
Also, using InputStreamResponseListener avoid keeping the response in-memory before sending it to the audiosink

Signed-off-by: Gwendal Roulleau <gwendal.roulleau@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java [hunk 1]
```diff
@@ -82,18 +90,16 @@ public class MimicTTSService implements TTSService {
...
-    /**
-     * Logger.
-     */
-    private final Logger logger = LoggerFactory.getLogger(MimicTTSService.class);
-
     private final MimicConfiguration config = new MimicConfiguration();
 
     private final Gson gson = new GsonBuilder().create();
 
+    private HttpClient httpClient;
```
Comment (lolodomo) [inline bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:97]: Could be final

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java [hunk 2]
```diff
@@ -223,27 +230,46 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
...
+        // the requested text. Average speaker speed estimated to 10/second.
+        // Will use a safe margin multiplicator (x5) to accept very slow mimic server
+        // So the constant chosen is 5 * 10 = /2
+        int timeout = text.length() / 2;
+
+        // check response status and return AudioStream
+        Response response;
+        try {
+            response = inputStreamResponseListener.get(timeout, TimeUnit.SECONDS);
+            if (response.getStatus() == 200) {
```
Comment (lolodomo) [inline bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:261]: You can use `HttpStatus.OK_200`

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java [hunk 3]
```diff
@@ -223,27 +230,46 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
...
+                String lengthHeader = response.getHeaders().get(HttpHeader.CONTENT_LENGTH);
+                long length = Long.parseLong(lengthHeader);
+                return new InputStreamAudioStream(inputStreamResponseListener.getInputStream(), AUDIO_FORMAT, length);
+            } else {
+                String errorMessage = "Cannot get wav from mimic url " + urlTTS + " with HTTP response code "
```
Comment (lolodomo) [inline bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:266]: Looking what I have done in the remote openHAB binding, I was aborting the request when code was not OK,, something like:
```
response.abort(new Exception(response.getReason()));
```

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java (hunk 1 above)
Comment (dalgwen) [reply bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:97]: Done

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java (hunk 2 above)
Comment (dalgwen) [reply bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:261]: Done

File: bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java (hunk 3 above)
Comment (dalgwen) [reply bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java:266]: It seems like a good practice, especially with callback response / deferred inputstream. Done

TY for your review !

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**